src/third-party/imtui/imtui-impl-ncurses.cpp

```diff
@@ -34,93 +34,80 @@
 #include <string>
 #include <thread>
 
-namespace {
-    struct VSync {
-        VSync(double fps_active = 60.0, double fps_idle = 60.0) :
-            tStepActive_us(1000000.0/fps_active),
-            tStepIdle_us(1000000.0/fps_idle) {}
+static WINDOW *imtui_win = nullptr;
 
-        uint64_t tStepActive_us;
-        uint64_t tStepIdle_us;
-        uint64_t tLast_us = t_us();
-        uint64_t tNext_us = tLast_us;
+namespace
+{
+struct VSync {
+    VSync( double fps_active = 60.0, double fps_idle = 60.0 ) :
+        tStepActive_us( 1000000.0 / fps_active ),
+        tStepIdle_us( 1000000.0 / fps_idle ) {}
 
-        inline uint64_t t_us() const {
-            return std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count(); // duh ..
-        }
+    uint64_t tStepActive_us;
+    uint64_t tStepIdle_us;
+    uint64_t tLast_us = t_us();
+    uint64_t tNext_us = tLast_us;
 
-        inline void wait(bool active) {
-            uint64_t tNow_us = t_us();
+    inline uint64_t t_us() const {
+        return std::chrono::duration_cast<std::chrono::microseconds>
+               ( std::chrono::high_resolution_clock::now().time_since_epoch() ).count(); // duh ..
+    }
 
-            auto tStep_us = active ? tStepActive_us : tStepIdle_us;
-            auto tNextCur_us = tNext_us + tStep_us;
+    inline void wait( bool active ) {
+        uint64_t tNow_us = t_us();
 
-            wtimeout(stdscr, 0);
-            while (tNow_us < tNextCur_us - 100) {
-                if (tNow_us + 0.5*tStepActive_us < tNextCur_us) {
-                    int ch = wgetch(stdscr);
+        auto tStep_us = active ? tStepActive_us : tStepIdle_us;
+        auto tNextCur_us = tNext_us + tStep_us;
 
-                    if (ch != ERR) {
-                        ungetch(ch);
-                        tNextCur_us = tNow_us;
-                        wtimeout(stdscr, 1);
+        while( tNow_us < tNextCur_us - 100 ) {
+            if( tNow_us + 0.5 * tStepActive_us < tNextCur_us ) {
+                int ch = wgetch( imtui_win );
 
-                        return;
-                    }
-                }
+                if( ch != ERR ) {
+                    ungetch( ch );
+                    tNextCur_us = tNow_us;
 
-                std::this_thread::sleep_for(std::chrono::microseconds(
-                        std::min((uint64_t)(0.9*tStepActive_us),
-                                 (uint64_t) (0.9*(tNextCur_us - tNow_us))
-                                 )));
-
-                tNow_us = t_us();
+                    return;
+                }
```

Comment:
I wrapped the code into a namespace because there was a name collision with some code in CDDA with similar or the same name. I'm not in Linux right now, but I can check later. The indentation is because of the namespacing... is it preferable to just not indent the code for this?

Reply:
It's not just namespace indented. The code also has astyle-forced spacing (eg. `if ( ch != ERR ) {` instead of `if (ch != ERR) {`).

Reply:
im resolving this because this is so old, but if you want to revisit this we can. i was editing the file in VS so its very possible some indenting was changed (VS will enforce your indenting style when you type a closing bracket) but I don't think there's much to be done - the indentation was going to change because of the namespacing and I made lots of changes to this file.